cmd/soroban-rpc/internal/methods/get_ledgers.go

```diff
@@ -0,0 +1,251 @@
+package methods
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/creachadair/jrpc2"
+
+	"github.com/stellar/go/xdr"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/db"
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/ledgerbucketwindow"
+)
+
+// isStartLedgerWithinBounds checks whether the request start ledger/cursor is within the max/min ledger
+// for the current RPC instance.
+func isStartLedgerWithinBounds(startLedger uint32, ledgerRange ledgerbucketwindow.LedgerRange) bool {
+	return startLedger >= ledgerRange.FirstLedger.Sequence && startLedger <= ledgerRange.LastLedger.Sequence
+}
+
+// GetLedgersRequest represents the request parameters for fetching ledgers.
+type GetLedgersRequest struct {
+	StartLedger uint32             `json:"startLedger"`
+	Pagination  *PaginationOptions `json:"pagination,omitempty"`
+	Format      string             `json:"xdrFormat,omitempty"`
+}
+
+// validate checks the validity of the request parameters.
+func (req *GetLedgersRequest) validate(maxLimit uint, ledgerRange ledgerbucketwindow.LedgerRange) error {
+	switch {
+	case req.Pagination != nil && req.Pagination.Cursor != "":
+		if req.StartLedger != 0 {
+			return errors.New("startLedger and cursor cannot both be set")
+		}
+	case !isStartLedgerWithinBounds(req.StartLedger, ledgerRange):
+		return fmt.Errorf(
+			"start ledger must be between the oldest ledger: %d and the latest ledger: %d for this rpc instance",
+			ledgerRange.FirstLedger.Sequence,
+			ledgerRange.LastLedger.Sequence,
+		)
+	}
+
+	if req.Pagination != nil && req.Pagination.Limit > maxLimit {
+		return fmt.Errorf("limit must not exceed %d", maxLimit)
+	}
+
+	return IsValidFormat(req.Format)
+}
+
+// LedgerInfo represents a single ledger in the response.
+type LedgerInfo struct {
+	Hash            string `json:"hash"`
+	Sequence        uint32 `json:"sequence"`
+	LedgerCloseTime int64  `json:"ledgerCloseTime"`
```

Comment:
don't forget 64-bit js quirks!
```suggestion
	LedgerCloseTime int64  `json:"ledgerCloseTime,string"`
```